cmd/flagger/main.go

```diff
@@ -87,7 +89,14 @@ func main() {
 		logger.Fatalf("Error building example clientset: %s", err.Error())
 	}
 
-	flaggerInformerFactory := informers.NewSharedInformerFactory(flaggerClient, time.Second*30)
+	var flaggerInformerFactory informers.SharedInformerFactory
+	logger.Infof("Flagger Canary 's Watcher is on namespace %s", namespace)
+	if namespace == "all" {
```

Comment:
If the namespace is string empty then NewSharedInformerFactoryWithOptions will create an informer for all namespaces. Please replace the default `all` with `""` and you can delete the if condition.

See https://github.com/kubernetes/client-go/blob/master/informers/factory.go#L110 and https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L294